spec/shared_examples/faraday_connection.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+RSpec.shared_examples "Faraday::Connection configuration block" do
```

Comment:
WDYTA accepting a block argument here (just so the outer specs don't _have_ to name it `connection`?
```suggestion
RSpec.shared_examples "Faraday::Connection configuration block" do |connection|
```

Then specs can call:
```
include_examples "Faraday::Connection configuration block", my_custom_faraday_connection
```

Reply:
The issue is that `my_custom_faraday_connection` (in your example) is generally defined in a `let` block in the outer specs.

When I try to pass a parameter to the `shared_examples` block, I get the following error message:

```
An error occurred while loading ./spec/iron_bank/authentications/token_spec.rb.
Failure/Error: include_examples "Faraday::Connection configuration block", connection
  `connection` is not available on an example group (e.g. a `describe` or `context` block). It is only available from within individual examples (e.g. `it` blocks) or from constructs that run in the scope of an example (e.g. `before`, `let`, etc).
# ./spec/iron_bank/authentications/token_spec.rb:152:in `block in <top (required)>'
# ./spec/iron_bank/authentications/token_spec.rb:6:in `<top (required)>
```

While I agree that it would be better to not have the outer specs explicitly provide a `connection` variable, I'd consider this an OK exception for the sake of readability. Another argument is that the outer spec still **include**  the shared examples, so they have to provide it with the expected variables.